### DISCUSSION TITLE
UnicodeEncodeError

### DISCUSSION THREAD

**Discussion by cascafico** [markqvist/Reticulum#613] [Nomad Network]
Hello.
After a modification in config [node] section I had the following error:

```
pi@rpi3:~/.nomadnetwork $ /home/pi/.local/bin/nomadnet 
Traceback (most recent call last):
  File "/home/pi/.local/bin/nomadnet", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/home/pi/.local/lib/python3.11/site-packages/nomadnet/nomadnet.py", line 51, in main
    program_setup(configarg, rnsconfigarg, daemon, console)
  File "/home/pi/.local/lib/python3.11/site-packages/nomadnet/nomadnet.py", line 11, in program_setup
    app = nomadnet.NomadNetworkApp(
          ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/pi/.local/lib/python3.11/site-packages/nomadnet/NomadNetworkApp.py", line 348, in __init__
    self.node = nomadnet.Node(self)
                ^^^^^^^^^^^^^^^^^^^
  File "/home/pi/.local/lib/python3.11/site-packages/nomadnet/Node.py", line 29, in __init__
    self.register_files()
  File "/home/pi/.local/lib/python3.11/site-packages/nomadnet/Node.py", line 81, in register_files
    self.destination.register_request_handler(
  File "/home/pi/.local/lib/python3.11/site-packages/RNS/Destination.py", line 385, in register_request_handler
    path_hash = RNS.Identity.truncated_hash(path.encode("utf-8"))
                                            ^^^^^^^^^^^^^^^^^^^^
UnicodeEncodeError: 'utf-8' codec can't encode character '\udce9' in position 28: surrogates not allowed

```
The config editing was just about two rows (files and pages paths). After this error I removed the lines, but it didn't solve the issue.

Some time ago I solved restarting nomadnet with node disabled, then restarting enabled again. This time it didn't work.

Maybe the following reticulum log could be related:
`[Error] Could not save path table to storage, the contained exception was: dictionary changed size during iteration`

**Comment by markqvist** [comment]
What operating system was this on? Any weird characters in your file names or paths?

**Reply by cascafico** [reply]
Linux rpi3 6.6.51+rpt-rpi-v7 #1 SMP Raspbian 1:6.6.51-
1+rpt3 (2024-10-08) armv7l GNU/Linux

The path I had set contained a sym link to a target folder with filenames containing spaces and parentheses. Please note that I removed those two lines (pages and files) before attempting to restart.



Edit:
I forgot to mention that I have rnsd running as a service (which does not show any errors in log). I guess nomadnet share its instance.

**Comment by cascafico** [comment]
> [Error] Could not save path table to storage, the contained exception was: dictionary changed size during iteration

Eventually I solved the UnicodeEncodeError that didn't allow me to run nomadnet removing any symlink from files folder.
